integration/qbft/tests/utils.go

```
@@ -0,0 +1,178 @@
+	"testing"
+	"time"
+)
+
+type Delay time.Duration
+type Slot phase0.Slot
+
+const (
+	NoDelay       = Delay(0)
+	OneRoundDelay = Delay(2 * time.Second)
+
+	DefaultSlot = Slot(phase0.Slot(spectestingutils.TestingDutySlot + 0)) //ZeroSlot
+)
+
+var (
+	KeySet4Committee  = spectestingutils.Testing4SharesSet()
+	KeySet7Committee  = spectestingutils.Testing7SharesSet()
+	KeySet10Committee = spectestingutils.Testing10SharesSet()
+	KeySet13Committee = spectestingutils.Testing13SharesSet()
+)
+
+type DutyProperties struct {
+	Slot  Slot
+	Idx   phase0.ValidatorIndex
+	Delay Delay
+}
+
+type StoredInstanceProperties struct {
+	Height specqbft.Height
+}
+
+func newStores(logger *zap.Logger) *qbftstorage.QBFTStores {
+	db, err := storage.GetStorageFactory(basedb.Options{
+		Type:   "badger-memory",
+		Path:   "",
+		Logger: zap.L(),
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	store := qbftstorage.New(db, logger, "integration-tests", protocolforks.GenesisForkVersion)
+
+	stores := qbftstorage.NewStores()
+	stores.Add(spectypes.BNRoleAttester, store)
+	stores.Add(spectypes.BNRoleProposer, store)
+	stores.Add(spectypes.BNRoleAggregator, store)
+	stores.Add(spectypes.BNRoleSyncCommittee, store)
+	stores.Add(spectypes.BNRoleSyncCommitteeContribution, store)
+
+	return stores
+}
+
+func createDuty(pk []byte, slot phase0.Slot, idx phase0.ValidatorIndex, role spectypes.BeaconRole) *spectypes.Duty {
+	var pkBytes [48]byte
+	copy(pkBytes[:], pk)
+
+	var testingDuty *spectypes.Duty
+	switch role {
+	case spectypes.BNRoleAttester:
+		testingDuty = spectestingutils.TestingAttesterDuty
+	case spectypes.BNRoleAggregator:
+		testingDuty = spectestingutils.TestingAggregatorDuty
+	case spectypes.BNRoleProposer:
+		testingDuty = spectestingutils.TestingProposerDuty
+	case spectypes.BNRoleSyncCommittee:
+		testingDuty = spectestingutils.TestingSyncCommitteeDuty
+	case spectypes.BNRoleSyncCommitteeContribution:
+		testingDuty = spectestingutils.TestingSyncCommitteeContributionDuty
```

please add a default case which should probably return an error